guides/plugins/plugins/storefront/add-scss-variables.md

```diff
@@ -0,0 +1,212 @@
+    {
+        // Will render: $sass-plugin-header-bg-color: "#59ccff";
+        $event->addVariable('sass-plugin-header-bg-color', '#59ccff');
+    }
+}
+```
+{% endtab %}
+{% tab title="<plugin root>/src/Resources/config/services.xml" %}
+```xml
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    
+    <services>
+        <service id="Swag\BasicExample\Subscriber\ThemeVariablesSubscriber">
+            <tag name="kernel.event_subscriber"/>
+        </service>
+    </services>
+</container>
+```
+{% endtab %}
+{% endtabs %}
+
+The `ThemeCompilerEnrichScssVariablesEvent` provides the `addVariable()` method which takes the following parameters:
+
+* `$name:` (string): The name of the SCSS variable. The passed string will be exactly in your SCSS so please be careful with special characters. We recommend using kebab-case here. The variable prefix `$` will be added automatically. We also recommend prefixing your variable name with your plugin's or company's name to prevent naming conflicts.
+* `$value:` (string): The value which should be assigned to the SCSS variable.
+* `$sanitize` (bool - optional): Optional parameter to remove special characters from the variables value. The parameter will also add quotes around the variables value. In most cases quotes are not needed e.g. for color hex values. But there may be situations where you want to pass individual strings to your SCSS variable.
+
+{% hint style="info" %}
+Please note that plugins are not sales-channel specific. Your SCSS variables are directly added in the SCSS compilation process and will be globally available throughout all themes and storefront sales-channels. If you want to change a variables value for each sales-channel you should use plugin config fields and follow the next example.
+{% endhint %}
+
+## Plugin config values as SCSS variables
+
+Inside your `ThemeVariablesSubscriber.php` you can also read values from the plugin configuration and assign those to a SCSS variable. This makes it also possible to have different values for each sales-channel. Depending on the selected sales-channel inside the plugin configuration in the administration.
+
+First of all lets add a new plugin configuration field according to the [Plugin Configurations](../plugin-fundamentals/add-plugin-configuration.md):
+
+{% code title="<plugin root>/src/Resources/config/config.xml" %}
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/shopware/platform/trunk/src/Core/System/SystemConfig/Schema/config.xsd">
+
+    <card>
+        <title>Example configuration</title>
+        <input-field>
```

Comment:
I suggest to have an example that uses the sw-colorpicker component. This can also be a good showcase how to use "custom" admin components for plugin configurations.

Reply:
@JoshuaBehrens - Thank you for the advice. I added the type `colopicker` in this example.

Reply:
Looks good :) I like very mucho